components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-node/reporting-node.component.ts

```diff
@@ -46,19 +51,35 @@ export class ReportingNodeComponent implements OnInit, OnDestroy {
     const reportQuery = this.reportQueryService.getReportQuery();
     reportQuery.filters = reportQuery.filters.concat([{type: {name: 'node_id'}, value: {id}}]);
 
-    this.statsService.getReports(reportQuery, {sort: 'latest_report.end_time', order: 'DESC'})
-    .pipe(takeUntil(this.isDestroyed))
-    .subscribe(reports => {
-      this.reports = reports;
-      const queryForReport = this.reportQueryService.getReportQueryForReport(reports[0]);
-      this.statsService.getSingleReport(reports[0].id, queryForReport)
-        .pipe(takeUntil(this.isDestroyed))
-        .subscribe(data => {
-          this.reportLoading = false;
-          this.layoutFacade.ShowPageLoading(false);
-          this.activeReport = Object.assign(reports[0], data);
-        });
+    this.page$.subscribe(page => {
+      this.page = page;
+      this.statsService.getReportsWithPages(reportQuery,
+        {sort: 'latest_report.end_time', order: 'DESC'}, page, this.pageSize)
+      .pipe(takeUntil(this.isDestroyed))
+      .subscribe(reportCollection => {
+        this.totalReports = reportCollection.totalReports;
+        this.reports = reportCollection.reports;
+
+        // load the first report only when first loading the page
+        if (!this.firstReportIsLoaded) {
+          this.firstReportIsLoaded = true;
+          const queryForReport = this.reportQueryService.getReportQueryForReport(this.reports[0]);
+          this.statsService.getSingleReport(this.reports[0].id, queryForReport)
+            .pipe(takeUntil(this.isDestroyed))
+            .subscribe(data => {
```

Comment:
Is it possible to give this `data` a type? Right now it is `any`.

Reply:
There is currently no type for this data. It is a report object that looks to have a deep structure. I am scared to start to create my own type for this without a large task.